Godeps/_workspace/src/k8s.io/kubernetes/pkg/apiserver/apiserver.go

```diff
@@ -177,7 +177,11 @@ func serviceErrorHandler(requestResolver *APIRequestInfoResolver, apiVersions []
 		}
 	}
 
-	errorJSON(apierrors.NewGenericServerResponse(serviceErr.Code, "", "", "", "", 0, false), codec, response.ResponseWriter)
+	resourceOrKind := requestInfo.Kind
+	if len(resourceOrKind) == 0 {
+		resourceOrKind = requestInfo.Resource
```

Comment:
Not sure I like putting a resource in the `Kind` field of a `StatusDetails`... they're not really interchangeable. I wish there was a way to get it into the message without doing that